tools/src/commands/GenerateBareApp.ts

```diff
@@ -187,13 +189,16 @@ async function symlinkPackages({
   }
 }
 
-async function updateRNVersion({ projectDir }: { projectDir: string }) {
+async function updateRNVersion({ projectDir, rnVersion }: { projectDir: string, rnVersion?: string }) {
+  if (!rnVersion) {
+    const mainPkg = require(path.resolve(EXPO_DIR, 'package.json'));
+    const rnVersionOnMain = mainPkg.resolutions['react-native'];
```

Comment:
We can't assume the `resolutions` object is there 😉 
```suggestion
    const rnVersionOnMain = mainPkg.resolutions?.['react-native'];
```